Basic/ClientDriven/Assets/Scripts/NetworkObjectSpawner.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.Networking;
+using Random = System.Random;
+
+public class NetworkObjectSpawner : NetworkBehaviour
+{
+    [SerializeField] NetworkObject prefabReference;
+
+    public void Start()
+    {
+        NetworkManager.Singleton.OnServerStarted += OnServerStartedIngredientSpawn;
```

Comment:
One thing here is to make sure to check NetworkManager.Singleton is not null before accessing it.

And another good pattern is to add the de-registration to this callback, on OnDestroy().